hazelcast/src/main/java/com/hazelcast/query/impl/predicates/PredicateUtils.java

```diff
@@ -45,4 +56,105 @@ public static int estimatedSizeOf(Collection<QueryableEntry> result) {
         }
         return result.size();
     }
+
+    /**
+     * @return {@code true} if the given value is considered as a null-like by
+     * predicates and indexes, {@code false} otherwise.
+     */
+    public static boolean isNull(Comparable value) {
+        return value == null || value == NULL;
+    }
+
+    /**
+     * @return {@code true} if the given predicate is a {@link RangePredicate},
+     * {@code false} otherwise.
+     */
+    public static boolean isRangePredicate(Predicate predicate) {
+        // XXX: NotEqualPredicate is a subclass of EqualPredicate
+        return predicate instanceof RangePredicate && !(predicate instanceof NotEqualPredicate);
+    }
+
+    /**
+     * @return {@code true} if the given predicate is an {@link EqualPredicate},
+     * {@code false} otherwise.
+     */
+    public static boolean isEqualPredicate(Predicate predicate) {
+        // XXX: NotEqualPredicate is a subclass of EqualPredicate
+        return predicate instanceof EqualPredicate && !(predicate instanceof NotEqualPredicate);
+    }
+
+    /**
+     * Produces canonical attribute representation by stripping an unnecessary
+     * "this." qualifier from the passed attribute, if any.
+     *
+     * @param attribute the attribute to canonicalize.
+     * @return the canonicalized attribute representation.
+     * @see #constructCanonicalCompositeIndexName
+     */
+    public static String canonicalizeAttribute(String attribute) {
+        return THIS_PATTERN.matcher(attribute).replaceFirst("");
+    }
+
+    /**
+     * Parses the given index name into components.
+     *
+     * @param name the index name to parse.
+     * @return the parsed components or {@code null} if the given index name
+     * doesn't describe a composite index components.
+     * @throws IllegalArgumentException if the given index name is empty.
+     * @throws IllegalArgumentException if the given index name contains empty
+     *                                  components.
+     * @throws IllegalArgumentException if the given index name contains
+     *                                  duplicate components.
+     * @see #constructCanonicalCompositeIndexName
+     */
+    public static String[] parseCompositeIndexComponents(String name) {
```

Comment:
Maybe something like `parseIndexName` or `parseToCompositeIndexComponents` can be clearer since index name is parsed to components here.